### PR TITLE
#8132 create person when creating a new case or a new contact

### PR DESCRIPTION
**Pull request by marius2301**
Fixes #

### CODE REVIEW

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/CoreAndPersonDto.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+	public PersonDto getPerson() {
+		return person;
+	}
+
+	public T getCoreData() {
+		return coreData;
+	}
+
+	public CoreAndPersonDto(PersonDto person, T coreData) {
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/caze/CoreAndPersonDto.java:20]: Constructors should be above the getters

File: sormas-api/src/main/java/de/symeda/sormas/api/contact/ContactAndPersonDto.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package de.symeda.sormas.api.contact;
+
+import de.symeda.sormas.api.person.PersonDto;
+
+public class ContactAndPersonDto extends ContactDto {
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/contact/ContactAndPersonDto.java:5]: I believe this class is no longer needed?

File: sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactFacadeEjb.java [hunk 3]
```diff
@@ -309,6 +311,13 @@ private ContactDto getContactWithoutPseudonyimizationByUuid(String uuid) {
 
 	@Override
 	public ContactDto save(@Valid @NotNull ContactDto dto) {
+		if (dto instanceof ContactAndPersonDto) {
```
Comment (MateStrysewske) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactFacadeEjb.java:314]: This code is probably no longer needed

File: sormas-rest/src/main/java/de/symeda/sormas/rest/CaseResource.java [hunk 4]
```diff
@@ -88,6 +89,12 @@
 		return savePushedDto(dtos, FacadeProvider.getCaseFacade()::save);
 	}
 
+	@POST
+	@Path("/post")
```
Comment (MateStrysewske) [inline sormas-rest/src/main/java/de/symeda/sormas/rest/CaseResource.java:93]: Please rename the endpoint to `pushWithPerson` (to be in line with the existing POST endpoint)

File: sormas-rest/src/main/java/de/symeda/sormas/rest/ContactResource.java [hunk 5]
```diff
@@ -84,11 +88,17 @@
 	@POST
 	@Path("/push")
 	public List<PushResult> postContacts(@Valid List<ContactDto> dtos) {
-
 		List<PushResult> result = savePushedDto(dtos, FacadeProvider.getContactFacade()::save);
 		return result;
 	}
 
+	@POST
+	@Path("/post")
```
Comment (MateStrysewske) [inline sormas-rest/src/main/java/de/symeda/sormas/rest/ContactResource.java:96]: Same as above

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

**Review by MateStrysewske — APPROVED** [review]

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
Merge conflicts :) looks good otherwise

**Review by BarnaBartha — APPROVED** [review]

**Pull request merged**